sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -522,9 +522,9 @@ def upload_blob_from_url(self, source_url, **kwargs):
         :param str source_url:
             A URL of up to 2 KB in length that specifies a file or blob.
             The value should be URL-encoded as it would appear in a request URI.
-            If the source is in another account, the source must either be public
-            or must be authenticated via a shared access signature. If the source
-            is public, no authentication is required.
+            The source must either be public or must be authenticated via a shared access signature,
+            regardless of whether the source is on the same or a different account.
```

Comment:
nit: Could remove this sentence and that implies that it is regardless of whether it's on the same account or not- will allow Jacob's feedback to trump mine here though. Otherwise this is good too!

Reply:
Sounds good!

Reply:
Hmm this isn't entirely true because this API also supports `source_authorization` which allows you to authenticate the source using a Bearer token (OAuth). The REST docs don't really reflect that, but we could here. How about "The source must either be public, must be authenticated via a shared access signature as part of the url, or using the `source_authorization` keyword." (I did drop the part about the accounts like Vincent suggested, also because the source doesn't have to come from a Storage account at all!)

Reply:
This looks good, will make the change :)